src/kbmod/work_unit.py

```diff
@@ -541,6 +548,237 @@ def to_fits(self, filename, overwrite=False):
+            config = SearchConfiguration.from_hdu(primary["kbmod_config"])
+
+            # Read in the global WCS from extension 0 if the information exists.
+            # We filter the warning that the image dimension does not match the WCS dimension
+            # since the primary header does not have an image.
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore", AstropyWarning)
+                global_wcs = extract_wcs_from_hdu_header(primary[0].header)
+
+            # Read the size and order information from the primary header.
+            num_images = primary[0].header["NUMIMG"]
+            n_constituents = primary[0].header["NCON"]
+            expected_num_images = (4 * num_images) + (2 * n_constituents) + 3
+
+            # Misc. reprojection metadata
+            reprojected = primary[0].header["REPRJCTD"]
+            heliocentric_distance = primary[0].header["HELIO"]
+            geocentric_distances = []
+            for i in range(num_images):
+                geocentric_distances.append(primary[0].header[f"GEO_{i}"])
+
+            per_image_wcs = []
+            per_image_ebd_wcs = []
+            constituent_images = []
+            for i in range(n_constituents):
+                # Extract the per-image WCS if one exists.
+                per_image_wcs.append(extract_wcs_from_hdu_header(primary[f"WCS_{i}"].header))
+                per_image_ebd_wcs.append(extract_wcs_from_hdu_header(primary[f"EBD_{i}"].header))
+                constituent_images.append(primary[f"WCS_{i}"].header["ILOC"])
+        per_image_indices = []
+        file_paths = []
+        for i in range(num_images):
+            shard_path = os.path.join(directory, f"{i}_{filename}")
+            if not Path(shard_path).is_file():
+                raise ValueError(f"No shard provided for index {i} for {filename}")
+            with fits.open(shard_path) as hdul:
+                # Read in the image file.
+                sci_hdu = hdul[f"SCI_{i}"]
+
+                # Read in the layered image from different extensions.
+                if not lazy:
+                    img = LayeredImage(
+                        sci_hdu.data.astype(np.single),
+                        hdul[f"VAR_{i}"].data.astype(np.single),
+                        hdul[f"MSK_{i}"].data.astype(np.single),
+                        PSF(hdul[f"PSF_{i}"].data),
+                        sci_hdu.header["MJD"],
+                    )
+
+                    # force_move destroys img object, but avoids a copy.
+                    im_stack.append_image(img, force_move=True)
+                else:
+                    file_paths.append(shard_path)
+
+                n_indices = sci_hdu.header["NIND"]
+                sub_indices = []
+                for j in range(n_indices):
+                    sub_indices.append(sci_hdu.header[f"IND_{j}"])
+                per_image_indices.append(sub_indices)
+
+        file_paths = None if lazy else file_paths
```

Comment:
This looks like it is backwards to me. I think we would want to make the `file_paths` `None` if the setting is not lazy, because we need `file_paths` for the later load.